### PR TITLE
update microsoft-azure-storage-explorer to 1.14.0

### PR DESCRIPTION
**Pull request by chriswl**
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

### CODE REVIEW

File: Casks/microsoft-azure-storage-explorer.rb [hunk 1]
```diff
@@ -1,8 +1,10 @@
 cask 'microsoft-azure-storage-explorer' do
-  version :latest
-  sha256 :no_check
+  version '1.14.0'
+  sha256 '6b23316dd85b5ba9621e747a00e54a073dcbb920b89ae72afa8b9db07915950d'
 
-  url 'https://download.microsoft.com/download/A/E/3/AE32C485-B62B-4437-92F7-8B6B2C48CB40/StorageExplorer_signed.zip'
+  # github.com was verified as official when first introduced to the cask
+  url 'https://github.com/microsoft/AzureStorageExplorer/releases/download/v1.14.0/Mac_StorageExplorer.zip'
```
Comment (miccal) [inline Casks/microsoft-azure-storage-explorer.rb:6]: Please change these two lines to:
```
# github.com/microsoft/AzureStorageExplorer/ was verified as official when first introduced to the cask
url "https://github.com/microsoft/AzureStorageExplorer/releases/download/v#{version}/Mac_StorageExplorer.zip"
```

Comment (chriswl) [reply Casks/microsoft-azure-storage-explorer.rb]: Thanks!

File: Casks/microsoft-azure-storage-explorer.rb [hunk 2]
```diff
@@ -1,8 +1,10 @@
 cask 'microsoft-azure-storage-explorer' do
-  version :latest
-  sha256 :no_check
+  version '1.14.0'
+  sha256 '6b23316dd85b5ba9621e747a00e54a073dcbb920b89ae72afa8b9db07915950d'
 
-  url 'https://download.microsoft.com/download/A/E/3/AE32C485-B62B-4437-92F7-8B6B2C48CB40/StorageExplorer_signed.zip'
+  # github.com/microsoft/AzureStorageExplorer/ was verified as official when first introduced to the cask
+  url "https://github.com/microsoft/AzureStorageExplorer/releases/download/v#{version}/Mac_StorageExplorer.zip"
```
Comment (vitorgalvao) [inline Casks/microsoft-azure-storage-explorer.rb:6]: How does one get here from the homepage?

Comment (chriswl) [reply Casks/microsoft-azure-storage-explorer.rb:6]: It's well hidden, but the [release notes](https://docs.microsoft.com/en-us/azure/vs-azure-tools-storage-explorer-relnotes) are linked from the homepage and have a link to the github releases page. Also, the github releases are from the official Microsoft org.

Comment (core-code) [reply Casks/microsoft-azure-storage-explorer.rb:6]: that sounds good enough :)

**Pull request merged**